### ISSUE TITLE
Parse error on java try/catch blocks

### ISSUE THREAD

**Issue by pplam** [github/semantic#235]
When  parsing `java` code contains `try/catch` blocks, `semantic` output some warning as following
~~~
➜ semantic parse --json Bar.java > tree.json        
time=2019-08-19T23:51:47Z+0800 msg="<filtered>:3:5-7:6: error: expected TryWithResourcesStatement, but got Block\n\n" level=Warning task="assign" github_request_id="-" github_is_public="False" path="<filtered>" language="Java" app="semantic" pid="28301" hostname="deMacBook.local" sha="development"
~~~
Given the input `Bar.java` code
~~~java
public class Bar {
  public void foo() {
    try {
      System.out.println(1/0);
    } catch (Exception e) {
      e.printStackTrace(System.out);
    }
  }
}
~~~
And the result `tree.json`
~~~json
{  
   "trees":[  
      {  
         "tree":{  
            "term":"Statements",
            "statements":[  
               {  
                  "term":"Class",
                  "classBody":{  
                     "children":[  
                        {  
                           "term":"Method",
                           "methodAccessControl":"Public",
                           "methodBody":{  
                              "children":[  
                                 {  
                                    "term":"Error",
                                    "errorActual":"Block",
                                    "errorCallStack":[  

                                    ],
                                    "errorChildren":[  

                                    ],
                                    "errorExpected":[  
                                       "TryWithResourcesStatement"
                                    ],
                                    "sourceRange":[  
                                       45,
                                       152
                                    ],
                                    "sourceSpan":{  
                                       "start":[  
                                          3,
                                          5
                                       ],
                                       "end":[  
                                          7,
                                          6
                                       ]
                                    }
                                 }
                              ],
                              "sourceRange":[  
                                 39,
                                 156
                              ],
                              "sourceSpan":{  
                                 "start":[  
                                    2,
                                    21
                                 ],
                                 "end":[  
                                    8,
                                    4
                                 ]
                              }
                           },
                           "methodContext":[  
                              {  
                                 "term":"Void",
                                 "sourceRange":[  
                                    28,
                                    32
                                 ],
                                 "sourceSpan":{  
                                    "start":[  
                                       2,
                                       10
                                    ],
                                    "end":[  
                                       2,
                                       14
                                    ]
                                 }
                              },
                              {  
                                 "term":"AccessibilityModifier",
                                 "contents":"public",
                                 "sourceRange":[  
                                    21,
                                    27
                                 ],
                                 "sourceSpan":{  
                                    "start":[  
                                       2,
                                       3
                                    ],
                                    "end":[  
                                       2,
                                       9
                                    ]
                                 }
                              }
                           ],
                           "methodName":{  
                              "term":"Identifier",
                              "name":"foo",
                              "sourceRange":[  
                                 33,
                                 36
                              ],
                              "sourceSpan":{  
                                 "start":[  
                                    2,
                                    15
                                 ],
                                 "end":[  
                                    2,
                                    18
                                 ]
                              }
                           },
                           "methodParameters":[  

                           ],
                           "methodReceiver":{  
                              "term":"Empty",
                              "sourceRange":[  
                                 28,
                                 28
                              ],
                              "sourceSpan":{  
                                 "start":[  
                                    2,
                                    10
                                 ],
                                 "end":[  
                                    2,
                                    10
                                 ]
                              }
                           },
                           "sourceRange":[  
                              21,
                              156
                           ],
                           "sourceSpan":{  
                              "start":[  
                                 2,
                                 3
                              ],
                              "end":[  
                                 8,
                                 4
                              ]
                           }
                        }
                     ],
                     "sourceRange":[  
                        17,
                        158
                     ],
                     "sourceSpan":{  
                        "start":[  
                           1,
                           18
                        ],
                        "end":[  
                           9,
                           2
                        ]
                     }
                  },
                  "classContext":[  
                     {  
                        "term":"AccessibilityModifier",
                        "contents":"public",
                        "sourceRange":[  
                           0,
                           6
                        ],
                        "sourceSpan":{  
                           "start":[  
                              1,
                              1
                           ],
                           "end":[  
                              1,
                              7
                           ]
                        }
                     }
                  ],
                  "classIdentifier":{  
                     "term":"Identifier",
                     "name":"Bar",
                     "sourceRange":[  
                        13,
                        16
                     ],
                     "sourceSpan":{  
                        "start":[  
                           1,
                           14
                        ],
                        "end":[  
                           1,
                           17
                        ]
                     }
                  },
                  "classSuperclasses":[  

                  ],
                  "sourceRange":[  
                     0,
                     158
                  ],
                  "sourceSpan":{  
                     "start":[  
                        1,
                        1
                     ],
                     "end":[  
                        9,
                        2
                     ]
                  }
               }
            ],
            "sourceRange":[  
               0,
               159
            ],
            "sourceSpan":{  
               "start":[  
                  1,
                  1
               ],
               "end":[  
                  10,
                  1
               ]
            }
         },
         "path":"Bar.java",
         "language":"Java"
      }
   ]
}
~~~
Notice that the body of method `foo` is of term "Error" in the result json, so there should be some issues in semantic on this case. Please help me fix that, thanks very much!

**Comment by pplam** [comment]
@patrickt I tried tree-sitter-java, it works fine, the result:
~~~
(program [0, 0] - [10, 0]
  (class_declaration [0, 0] - [8, 1]
    (modifiers [0, 0] - [0, 6])
    name: (identifier [0, 13] - [0, 16])
    body: (class_body [0, 17] - [8, 1]
      (method_declaration [1, 2] - [7, 3]
        (modifiers [1, 2] - [1, 8])
        type: (void_type [1, 9] - [1, 13])
        name: (identifier [1, 14] - [1, 17])
        parameters: (formal_parameters [1, 17] - [1, 19])
        body: (block [1, 20] - [7, 3]
          (try_statement [2, 4] - [6, 5]
            body: (block [2, 8] - [4, 5]
              (expression_statement [3, 6] - [3, 30]
                (method_invocation [3, 6] - [3, 29]
                  object: (scoped_identifier [3, 6] - [3, 16]
                    (identifier [3, 6] - [3, 12])
                    (identifier [3, 13] - [3, 16]))
                  name: (identifier [3, 17] - [3, 24])
                  arguments: (argument_list [3, 24] - [3, 29]
                    (binary_expression [3, 25] - [3, 28]
                      left: (decimal_integer_literal [3, 25] - [3, 26])
                      right: (decimal_integer_literal [3, 27] - [3, 28]))))))
            (catch_clause [4, 6] - [6, 5]
              (catch_formal_parameter [4, 13] - [4, 24]
                (catch_type [4, 13] - [4, 22]
                  (type_identifier [4, 13] - [4, 22]))
                name: (identifier [4, 23] - [4, 24]))
              body: (block [4, 26] - [6, 5]
                (expression_statement [5, 6] - [5, 36]
                  (method_invocation [5, 6] - [5, 35]
                    object: (identifier [5, 6] - [5, 7])
                    name: (identifier [5, 8] - [5, 23])
                    arguments: (argument_list [5, 23] - [5, 35]
                      (scoped_identifier [5, 24] - [5, 34]
                        (identifier [5, 24] - [5, 30])
                        (identifier [5, 31] - [5, 34])))))))))))))
~~~

**Comment by tclem** [comment]
From the original error, I believe this is an assignment issue in semantic. The `try_statement` is assigned [here](https://github.com/github/semantic/blob/master/src/Language/Java/Assignment.hs#L454-L455) and my first guess is that `symbol TryStatement *> children tryWithResources` is consuming the `TryStatement` too soon - we need to alternate between a normal try and a try with resource _after_ consuming `TryStatement`. The `Bar.java` would make a good fixture to add.

**Comment by patrickt** [comment]
Aha, I was wrong!

**Comment by robrix** [comment]
Fixed by #238.
